core/services/chainlink/config_legacy.go

```diff
@@ -0,0 +1,708 @@
+package chainlink
+
+import (
+	"net"
+	"net/url"
+	"path/filepath"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/gin-gonic/contrib/sessions"
+	uuid "github.com/satori/go.uuid"
+	"go.uber.org/zap/zapcore"
+
+	"github.com/smartcontractkit/libocr/commontypes"
+	ocrnetworking "github.com/smartcontractkit/libocr/networking"
+
+	coreconfig "github.com/smartcontractkit/chainlink/core/config"
+	v2 "github.com/smartcontractkit/chainlink/core/config/v2"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services/keystore/keys/ethkey"
+	"github.com/smartcontractkit/chainlink/core/services/keystore/keys/p2pkey"
+	"github.com/smartcontractkit/chainlink/core/store/dialects"
+	"github.com/smartcontractkit/chainlink/core/store/models"
+	"github.com/smartcontractkit/chainlink/core/utils"
+)
+
+// legacyGeneralConfig is a wrapper to adapt Config to the legacy config.GeneralConfig interface.
```

Comment:
Have we decided to keep legacy interfaces alongside new structs forever? Or are we later going to expose new config structs directly in the Client object and slowly deprecate legacy config interface calls?

Reply:
I'm not sure if this has been decided - we may need to see how things settle first, but I imagine the more granular interfaces will still be useful to keep around.

Reply:
@samsondav

Reply:
Having an interface that doesn't quite match the underlying structs might be a PITA to maintain and also confusing to anybody new reading the code.

Reply:
Oh yeah they will definitely evolve after the old config is removed - at least for trivial type changes but probably other ways too. I just meant that having interface abstractions rather than direct access could still be useful (e.g. testing, avoiding import cycles, swapping implementations, etc.).